clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -150,6 +152,8 @@ private void maybePropagateMetadataError() {
         try {
             metadata.maybeThrowAnyException();
         } catch (Exception e) {
+            metadataException.completeExceptionally(e);
+            metadataException = metadataException.newIncompleteFuture();
```

Comment:
What's the benefit of `newIncompleteFuture()` over `new CompletableFuture()`?

Reply:
I don’t see any real benefit; I simply chose to use this method because Java provides it. Ultimately, this method internally just creates a new CompletedFuture.
```java
    public <U> CompletableFuture<U> newIncompleteFuture() {
        return new CompletableFuture<U>();
    }
```